core/modules/replica/HttpIngestIndexModule.cc

```diff
@@ -0,0 +1,187 @@
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/HttpIngestIndexModule.h"
+
+// Qserv headers
+#include "replica/Configuration.h"
+#include "replica/DatabaseMySQL.h"
+#include "replica/DatabaseServices.h"
+#include "replica/HttpExceptions.h"
+#include "replica/IndexJob.h"
+#include "replica/ServiceProvider.h"
+
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+void HttpIngestIndexModule::process(Controller::Ptr const& controller,
+                                    string const& taskName,
+                                    HttpProcessorConfig const& processorConfig,
+                                    qhttp::Request::Ptr const& req,
+                                    qhttp::Response::Ptr const& resp,
+                                    string const& subModuleName,
+                                    HttpModule::AuthType const authType) {
+    HttpIngestIndexModule module(controller, taskName, processorConfig, req, resp);
+    module.execute(subModuleName, authType);
+}
+
+
+HttpIngestIndexModule::HttpIngestIndexModule(Controller::Ptr const& controller,
+                                             string const& taskName,
+                                             HttpProcessorConfig const& processorConfig,
+                                             qhttp::Request::Ptr const& req,
+                                             qhttp::Response::Ptr const& resp)
+    :   HttpModule(controller, taskName, processorConfig, req, resp) {
+}
+
+
+json HttpIngestIndexModule::executeImpl(string const& subModuleName) {
+    if (subModuleName == "BUILD-SECONDARY-INDEX") return _buildSecondaryIndex();
+    throw invalid_argument(
+            context() + "::" + string(__func__) +
+            "  unsupported sub-module: '" + subModuleName + "'");
+}
+
+
+json HttpIngestIndexModule::_buildSecondaryIndex() {
+    debug(__func__);
+
+    auto const databaseServices = controller()->serviceProvider()->databaseServices();
+    auto const config = controller()->serviceProvider()->config();
+
+    string const database = body().required<string>("database");
+    bool const allowForPublished = (bool)body().optional<int>("allow_for_published", 0);
```

Comment:
do we allow c-style casts? I'd expect to see `body().optional<int>("allow_for_published", 0) != 0` rather than `(bool)...`